x/profiles/types/models_chain_links.go

```diff
@@ -202,6 +204,57 @@ func (b Base58Address) VerifyPubKey(key cryptotypes.PubKey) (bool, error) {
 
 // --------------------------------------------------------------------------------------------------------------------
 
+var _ AddressData = &HexAddress{}
+
+// NewHexAddress returns a new HexAddress instance
+// NOTE: Currently it only supports keccak256-uncompressed addresses
+func NewHexAddress(value string) *HexAddress {
+	return &HexAddress{Value: value}
+}
+
+// Validate implements AddressData
+func (h HexAddress) Validate() error {
+	if strings.TrimSpace(h.Value) == "" {
+		return fmt.Errorf("value cannot be empty or blank")
+	}
+
+	if len(h.Value) <= 2 {
+		return fmt.Errorf("address cannot be smaller than prefix")
+	}
+
+	prefix, addrWithoutPrefix := h.Value[:2], h.Value[2:]
+	if prefix != "0x" {
+		return fmt.Errorf("prefix does not match")
```

Comment:
I think we should be more clear with this error, extending this with "address prefix does not match the 0x prefix"